### PR TITLE
 Prepare all queries to make type casting easier

### PR DESCRIPTION
**Pull request by whatyouhide**
This is what Postgrex and MyXQL do as well: they prepare all queries, so that the driver (Xandra in our case) has all the information it needs on types. This means we don't need to do the `{type, value}` dumping that caused issues with `nil` as well, which is awesome.

I found this out because we could not dump a `bigint` column, since `dumpers/2` would be called with just `:integer` and so we wouldn't know which kind of int that would be.

The tests are a bit of a mess right now, but if that's ok, we could merge this and then I'll clean them up.

By the way: Xandra caches prepared queries, so preparing and then executing does two round trips *only the first time* for each node.

### CODE REVIEW

File: .formatter.exs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Used by "mix format"
 [
+  import_deps: [:ecto, :ecto_sql],
```
Comment (whatyouhide) [inline .formatter.exs:3]: Unrelelated-but-nice™

File: config/config.exs [hunk 2]
```diff
@@ -6,5 +6,5 @@ if config_env() == :test do
     xandra_module: XandraMock
 
   config :logger,
-    level: :info
+    level: String.to_existing_atom(System.get_env("LOG_LEVEL", "info"))
```
Comment (whatyouhide) [inline config/config.exs:9]: Unrelelated-but-nice™

File: lib/exandra.ex [hunk 3]
```diff
@@ -33,83 +33,37 @@ defmodule Exandra do
...
-
-  def dumpers({:array, type}, dumper),
-    do: [&encode_array(Ecto.Type.embedded_dump(type, &1, dumper), type)]
-
-  def dumpers(:uuid, type), do: [type, &encode_uuid/1]
-
+  def dumpers(:binary_id, type), do: [type, Ecto.UUID]
+  def dumpers(:map, _type), do: [&Jason.encode/1]
+  def dumpers(:naive_datetime, _type), do: [&naive_datetime_to_datetime/1]
+  def dumpers({:map, _}, type), do: [type]
```
Comment (whatyouhide) [inline lib/exandra.ex:49]: Much simpler dumping like this

Comment (vinniefranco) [reply lib/exandra.ex:49]: This is a great simplification

**Review by vinniefranco — APPROVED** [review]
Amazing simplification

**Pull request merged**